commit 81705cd444387287452622d713ae83949184bdfc
Author: Krzysztof Opasiak <k.opasiak@samsung.com>
Date:   Mon Jan 16 08:40:57 2017 +0100

    usb: gadget: udc-core: Rescan pending list on driver unbind
    
    commit 8236800da115a3e24b9165c573067343f51cf5ea upstream.
    
    Since:
    
    commit 855ed04a3758 ("usb: gadget: udc-core: independent registration
    of gadgets and gadget drivers")
    
    if we load gadget module but there is no free udc available
    then it will be stored on a pending gadgets list.
    
    $ modprobe g_zero.ko
    $ modprobe g_ether.ko
    [] udc-core: couldn't find an available UDC - added [g_ether] to list
    of pending drivers
    
    We scan this list each time when new UDC appears in system.
    But we can get a free UDC each time after gadget unbind.
    This commit add scanning of that list directly after unbinding
    gadget from udc.
    
    Thanks to this, when we unload first gadget:
    
    $ rmmod g_zero.ko
    
    gadget which is pending is automatically
    attached to that UDC (if name matches).
    
    Fixes: 855ed04a3758  ("usb: gadget: udc-core: independent registration of gadgets and gadget drivers")
    Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>
    Signed-off-by: Felipe Balbi <felipe.balbi@linux.intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1311_81705cd444387287452622d713ae83949184bdfc.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1311_81705cd444387287452622d713ae83949184bdfc.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
